File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        empty = dict()
        for i in nums:
            if i in empty:
                empty[i] += 1
            else:
                empty[i] = 1
            if empty[i] > 1:
                return True
        return False

        

File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        return BSearch(0, nums.size()-1,nums,target);
    }

    int BSearch(int l, int r, vector<int>& nums, int target)
    {
        if(l>r) return -1;
        int m = l + (r-l) /2;

        if(nums[m] == target) return m;
        return (nums[m] < target) ? BSearch(m+1, r, nums,target) : BSearch(l, m - 1, nums,target);
    }
};
